lib/widgets/totals_table.dart

```diff
@@ -82,32 +95,60 @@ class TotalsTable extends StatelessWidget {
     for (var item in [...fodderItems, ...concentrateItems]) {
       for (var col in columns) {
         String key = col['key'] as String;
-        totals[key] = (totals[key] ?? 0) + (item[key] as num);
+        totals[key] = (totals[key] ?? 0) + (item[key] ?? 0);
+      }
+    }
+
+    double totalDM = totals['dmIntake']!;
+    totals['concentrateDM'] = 0;
+
+    // dmIntake totals for concentrate items
+    for (var item in concentrateItems) {
+      totals['concentrateDM'] =
+          (totals['concentrateDM'] ?? 0) + (item['concentrateDM'] ?? 0);
+    }
+
+    // Convert to %DM
+    List<String> percentageKeys = [
+      'concentrateIntake',
+      'cpIntake',
+      'ndfIntake',
+      'caIntake',
+      'pIntake'
+    ];
+    for (String key in percentageKeys) {
+      if (totalDM > 0) {
+        totals[key] = (totals[key]! / totalDM) * 100;
+      } else {
+        totals[key] = 0;
       }
     }
 
     return totals;
   }
 
   Widget _buildComparisonIcon(
-      double total, double requirement, bool isPercentage) {
-    const double percentageThreshold = 2;
-    const double absoluteThreshold = 0.05;
+      String key, double total, Map<String, dynamic> columnData) {
+    if (columnData['reqKey'] == null) {
+      return SizedBox.shrink(); // No comparison for this key
+    }
 
-    bool isWithinRange;
-    if (isPercentage) {
-      isWithinRange = (total - requirement).abs() <= percentageThreshold;
-    } else {
-      isWithinRange =
-          (total - requirement).abs() <= absoluteThreshold * requirement;
+    double min = columnData['min'].toDouble();
+    double max = columnData['max'].toDouble();
+
+    if (key == 'dmIntake' || key == 'meIntake') {
```

Comment:
Ah, I see you do multiply the values. I am not familiar with Dart, but would it make sense to give the correct limits when creating the object (eg. lines 27-28)